cli/installer/install-azd.ps1

```diff
@@ -315,106 +328,43 @@ try {
-                    reportTelemetryIfEnabled 'InstallFailed' 'FileCopyFailed'
-                    exit 1
-                }
-            } else {
-                Copy-Item "$tempFolder/decompress/$binFilename" $outputFilename  -ErrorAction Stop | Out-Null
-            }
-        } else {
-            if (!$SkipVerify) {
-                try {
-                    Write-Verbose "Verifying signature of $releaseArtifactFilename" -Verbose:$Verbose
-                    $signature = Get-AuthenticodeSignature $releaseArtifactFilename
-                    if ($signature.Status -ne 'Valid') {
-                        Write-Error "Signature of $releaseArtifactFilename is not valid"
-                        reportTelemetryIfEnabled 'InstallFailed' 'SignatureVerificationFailed'
-                        exit 1
-                    }
-                } catch {
-                    Write-Error "Could not verify signature of $releaseArtifactFilename"
-                    Write-Error $_
-                    reportTelemetryIfEnabled 'InstallFailed' 'SignatureVerificationFailed'
-                    exit 1
-                }
+        Write-Verbose "Installing MSI" -Verbose:$Verbose
+        $MSIEXEC = "${env:SystemRoot}\System32\msiexec.exe"
+        $installProcess = Start-Process $MSIEXEC `
+            -ArgumentList @("/i", $releaseArtifactFilename, "/qn", "INSTALLDIR=`"$InstallFolder`"", "INSTALLEDBY=`"install-azd.ps1`"") `
+            -PassThru `
+            -Wait
+
+        if ($installProcess.ExitCode) {
+            if ($installProcess.ExitCode -eq 1603) {
+                Write-Host "A later verison of Azure Developer CLI is already installed. Use 'Add or remove programs' to uninstall the current version and try again."
+                exit 0
             }
 
-            Write-Verbose "Installing MSI" -Verbose:$Verbose
-            $MSIEXEC = "${env:SystemRoot}\System32\msiexec.exe"
-            $installProcess = Start-Process $MSIEXEC `
-                -ArgumentList @("/i", $releaseArtifactFilename, "/qn", "INSTALLDIR=`"$InstallFolder`"") `
-                -PassThru `
-                -Wait
-
-            if ($installProcess.ExitCode) {
-                if ($installProcess.ExitCode -eq 1603) {
-                    Write-Host "A later verison of Azure Developer CLI is already installed. Use 'Add or remove programs' to uninstall the current version and try again."
-                    exit 0
-                }
-
-                Write-Error "Could not install MSI at $releaseArtifactFilename. msiexec.exe returned exit code: $($installProcess.ExitCode)"
+            Write-Error "Could not install MSI at $releaseArtifactFilename. msiexec.exe returned exit code: $($installProcess.ExitCode)"
 
-                reportTelemetryIfEnabled 'InstallFailed' 'MsiFailure'
-                exit 1
-            }
+            reportTelemetryIfEnabled 'InstallFailed' 'MsiFailure'
```

Comment:
You should telemeter the exit code. Most will be 1603, but there are a couple others that would be worth investigating. To get any more detail about 1603s, you need an actual chainer that can monitor internal messages. Alternatively, my `MSI` module can but I doubt you'd want an external dependency. I could maybe port the code, but it will still require a binary dependency.

Reply:
Thanks! I reworded the output about 1603 and removed the `exit`: 

```
"A later version of Azure Developer CLI may already be installed. Use 'Add or remove programs' to uninstall that version and try again."
``` 

That will result in updated telemetry